packages/core/contracts-ovm/insured-bridge/implementation/BridgePool.sol

```diff
@@ -1,12 +1,359 @@
+        newDeposit.depositState = DepositState.PendingSlow;
+        newDeposit.depositType = DepositType.Slow;
+        newDeposit.relayData = RelayAncillaryDataContents({
+            depositId: depositId,
+            l2Sender: l2Sender,
+            recipient: recipient,
+            depositTimestamp: depositTimestamp,
+            l1Token: l1Token,
+            amount: amount,
+            maxFeePct: maxFeePct,
+            proposerRewardPct: proposerRewardPct,
+            realizedFeePct: realizedFeePct,
+            slowRelayer: msg.sender
+        });
+        newDeposit.priceRequestAncillaryData = getRelayAncillaryData(newDeposit.relayData);
+
+        // Request a price for the relay identifier and propose "true" optimistically. These methods will pull the
+        // (proposer reward + proposer bond + final fee) from the caller.
+        // Note: We can't simply set the price request timestamp equal to the `depositTimestamp`, which is dependent
+        // on the L2 VM on which the DepositContract is deployed. Imagine if the timestamps on the L2 have an offset
+        // that are always "in the future" relative to L1 blocks, then the OptimisticOracle would always reject
+        // requests.
+        uint256 requestTimestamp = getCurrentTime();
+        _requestOraclePriceRelay(
+            l1Token,
+            amount,
+            requestTimestamp,
+            newDeposit.priceRequestAncillaryData,
+            proposerRewardPct
+        );
+        _proposeOraclePriceRelay(l1Token, amount, depositTimestamp, newDeposit.priceRequestAncillaryData);
+
+        // TODO: There is more data we'd like to emit, such as depositId, but we are limited by how many variables
+        // we can fit into an event. Perhaps we need multiple events? Or think more critically about what information
+        // clients will want.
+        emit DepositRelayed(
+            l2Sender,
+            depositTimestamp,
+            recipient,
+            l1Token,
+            msg.sender,
+            amount,
+            proposerRewardPct,
+            realizedFeePct,
+            bridgePoolFactory.depositContract()
+        );
+    }
+
+    function speedUpRelay(uint256 depositId) public {}
+
+    function finalizeRelay(uint256 depositId) public {}
+
+    function settleDisputedRelay(uint256 depositId, address slowRelayer) public {}
+
+    /**
+     * @notice Returns ancillary data containing all relevant Relay data that voters can format into UTF8 and use to
+     * determine if the relay is valid.
+     * @param _relayData Contains relevant relay data.
+     * @return bytes New ancillary data that can be decoded into UTF8.
+     */
+    function getRelayAncillaryData(RelayAncillaryDataContents memory _relayData) public view returns (bytes memory) {
+        bytes memory intermediateAncillaryData = "";
+
+        // Add relay data inferred from the original deposit on L2:
+        intermediateAncillaryData = AncillaryData.appendKeyValueUint(
```

Comment:
Could we hash _most_ of this information and just match it to a hash emitted on the deposit side? The only exception might be the fee that's computed by the relayer.

Just thinking about ways to avoid storing data on mainnet.

Reply:
We definitely could. @chrismaree